src/ensembl/src/shared/components/upload/Upload.test.tsx

```diff
@@ -30,88 +30,115 @@ const defaultProps = {
 
 const event = { preventDefault: noop, stopPropagation: noop };
 
-const renderUpload = (props: any) => mount(<Upload {...props} />);
+const renderUpload = (props: any) => render(<Upload {...props} />);
 
 const fileContents = faker.random.words();
 
 const generateFile = () => {
-  return new Blob([fileContents], { type: 'text/plain' });
+  return new File([fileContents], 'file.txt', { type: 'text/plain' });
 };
 
 const files = times(faker.random.number(10), () => generateFile());
-const addEventListener = jest.fn((_, evtHandler) => {
-  evtHandler();
-});
+
 const mockFileReader = {
-  addEventListener,
-  readAsText: jest.fn(),
+  onload: jest.fn(),
+  readAsText: jest.fn().mockImplementation(function () {
+    mockFileReader.onload();
+  }),
   readAsArrayBuffer: jest.fn(),
   readAsBinaryString: jest.fn(),
   readAsDataURL: jest.fn(),
   result: fileContents
 };
 
 describe('Upload', () => {
-  let wrapper: any;
+  let container: any;
```

Comment:
Similar to the comment for the BadgedButton — better run the `const { container } = renderUpload(defaultProps);` in each test. Also, it looks like the `defaultProps` can be moved inside the `renderUpload` function in the same manner it's done in other tests, and the type of the props passed to the `renderUpload` function can be the `Partial` of Upload component's props rather than `any`.